sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/PartitionKeyBuilder.java

```diff
@@ -0,0 +1,90 @@
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.implementation.Undefined;
+import com.azure.cosmos.implementation.routing.PartitionKeyInternal;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PartitionKeyBuilder {
+    private List<Object> partitionKeyValues;
+
+    /**
+     * Constructor. CREATE a new instance of the PartitionKeyBuilder object.
+     */
+    public PartitionKeyBuilder()
+    {
+        this.partitionKeyValues = new ArrayList<Object>();
+    }
+
+    public PartitionKeyBuilder Add(String value)
+    {
+        this.partitionKeyValues.add(value);
+        return this;
+    }
+
+    public PartitionKeyBuilder Add(double value)
+    {
+        this.partitionKeyValues.add(value);
+        return this;
+    }
+
+    public PartitionKeyBuilder Add(boolean value)
+    {
+        this.partitionKeyValues.add(value);
+        return this;
+    }
+
+    public PartitionKeyBuilder AddNullValue()
```

Comment:
This method does not seem right... may be there should be a "remove" method instead but trying to set a "null" as a legit property value, it will just complicate the serialization/de-serialization of certain payloads.

Reply:
This method is because we support adding a null value as a partition key value.